session/session.go

```diff
@@ -2302,6 +2302,25 @@ func (s *session) preparedStmtExec(ctx context.Context, execStmt *ast.ExecuteStm
 	}
 	sessionExecuteCompileDurationGeneral.Observe(time.Since(s.sessionVars.StartTime).Seconds())
 	logGeneralQuery(st, s, true)
+
+	pointShortPathOK, err := plannercore.IsPointPlanShortPathOK(s, prepareStmt)
+	if err != nil {
+		return nil, err
+	}
+	if pointShortPathOK {
```

Comment:
From https://github.com/pingcap/tidb/pull/36587/files#diff-10dab83c04a99358d431906d37f48b92be97d50a2719d4ddf81c62d3b03a0aebL2353

Reply:
I remembered that `case nil` is added intentionly, https://github.com/Reminiscent/tidb/commit/92b1b8e0e9598952917ab16fff0e17bcdac74992
Safe to remove?

Reply:
We can ignore this case-branch since `runStmt` is used by default.